Sources/PubNubChatComponents/Previews/SharedPreviewData.swift

```diff
@@ -0,0 +1,186 @@
+//
+//  File 2.swift
+//  
+//
+//  Created by Jakub Guz on 9/19/22.
+//
+
+#if canImport(SwiftUI) && DEBUG
+
+import Foundation
+import PubNub
+import PubNubChat
+import CoreData
+import Combine
+
+// The idea could be creating a shared class (or something similar) that holds shared data.
+// Then it's up to you what data you would like to use in your previews. Please don't treat existing implementation very serious, it's rather a POC 
+
+// MARK: ChatProvider with in-memory storage
+
+public let testChatProvider: PubNubChatProvider = {
```

Comment:
Correct me if I am wrong, but current design supports inheritance so you can make new TestChatProvider class. Is it safe to share testChatProvider between all of previews?

Reply:
It should be safe as long as provider remains immutable. It may not be possible though so we could separately create it for each preview.